static/js/video-apis/youtube/YoutubePlayer.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * Manages the state and interactions for an instance
+ * of the Youtube Iframe player.
+ */
+export default class YoutubePlayer {
+  /**
+   * @param {YT} YT the youtube iframe api
+   * @param {HTMLElement} el element to create the player for 
+   * @param {Function} onPlay runs when the play button is hit
+   */
+  constructor(YT, el, onPlay) {
+    this.player = new YT.Player(el, {
+      events: {
+        onStateChange: event => this.onStateChange(event)
+      }
+    });
+    this.playerState = YT.PlayerState.UNSTARTED;
+    this.onPlay = onPlay;
+  }
+
+  onStateChange({ data }) {
```

Comment:
If a video switched between states like buffering and playing, wouldn't this keep calling analyticsReporter to send a report that a video is played? Not sure if this would be an issue or something we want

Reply:
yup that's definitely something we need to handle! I missed this